pkg/common/adaptor/handler.go

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2022 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package adaptor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/cloudwego/hertz/pkg/app"
+	"github.com/cloudwego/hertz/pkg/common/hlog"
+	"github.com/cloudwego/hertz/pkg/protocol/consts"
+)
+
+// NewHertzHTTPHandlerFunc wraps net/http handler to hertz app.HandlerFunc,
+// so it can be passed to hertz server
+//
+// While this function may be used for easy switching from net/http to hertz,
+// it has the following drawbacks comparing to using manually written hertz
+// request handler:
+//
+//   - net/http -> hertz handler conversion has some overhead,
+//     so the returned handler will be always slower than manually written
+//     hertz handler.
+//
+// So it is advisable using this function only for net/http -> hertz switching.
+// Then manually convert net/http handlers to hertz handlers
+func NewHertzHTTPHandlerFunc(h http.HandlerFunc) app.HandlerFunc {
```

Comment:
Why would you pack an extra layer

Reply:
https://github.com/cloudwego/hertz/pull/253#discussion_r973595291
without this extra layer, caller should do like this:

    index := http.HandlerFunc(pprof.Index)  // convert
    prefixRouter.GET("/", NewHertzHTTPHandler(index))

this layer can help user to convert f(w http.ResponseWriter, r *http.Request) to  http.HandlerFunc

Reply:
thanks